trustmanager/keyfilestore.go

```diff
@@ -106,64 +110,122 @@ func (s *KeyMemoryStore) ListKeys() []string {
 	return listKeys(s)
 }
 
-func addKey(s LimitedFileStore, name string, privKey data.PrivateKey) error {
+// RemoveKey removes the key from the keystore
+func (s *KeyMemoryStore) RemoveKey(name string) error {
+	return removeKey(s, name)
+}
+
+func addKey(s LimitedFileStore, passphraseRetriever PassphraseRetriever, name, alias string, privKey data.PrivateKey) error {
 	pemPrivKey, err := KeyToPEM(privKey)
 	if err != nil {
 		return err
 	}
 
-	return s.Add(name, pemPrivKey)
-}
-
-func getKey(s LimitedFileStore, name string) (data.PrivateKey, error) {
-	keyBytes, err := s.Get(name)
-	if err != nil {
-		return nil, err
+	attempts := 0
+	passphrase := ""
+	giveup := false
+	for {
+		passphrase, giveup, err = passphraseRetriever(name, alias, true, attempts)
+		if err != nil {
+			attempts++
+			continue
+		}
+		if giveup {
+			return errors.New("obtaining passphrase failed")
+		}
+		if attempts > 10 {
```

Comment:
Rather than hardcoding 10 in multiple places, let's make this a constant at the top of the file.